### PR TITLE
feat: Run.sh new frontend support #452

### LINKED ISSUE #452: run.sh new frontend support
## Feature Request

Is your feature request related to an existing [Issue](https://www.github.com/Auto-DL/Generator/issues) or a new implementation?

No

### Feature Description

Parameter to build/run the frontend of choice
-f or --frontend with options: v1 and v2 (default)

on v1: same routine as now
v2: starts autodl-platform nextjs application

### PR DESCRIPTION
**Pull request by ashish-ad**
# Pull Request

## What does this PR do?
Fixes #452

This PR is related to a feature improvement in auto-dl for giving options for the installation of two front-end versions.
The motivation behind the PR is related to WOC2.0 participation.

## What part does this affect?
- [x] FrontEnd.
- [ ] BackEnd.
- [ ] Documentation.
- [x] Other. (Please specify below)

## Before submitting
- [x] Was this discussed/approved via a GitHub issue or slack?
- [x] Did you read the [contributor guideline](https://github.com/Auto-DL/Auto-DL/blob/v1-beta/CONTRIBUTING.md)?
- [x] Did you ensure that there aren't any other open [Pull Requests](https://www.github.com/Auto-DL/Generator/pulls) for the same update/change?
- [x] Did you make sure the title is self-explanatory and the description concisely explains the PR?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure the code is clean and docstrings have been added or updated as required?
- [x] Did you make sure the code is linted/formatted locally prior to submission? (using `black` and/or `prettier`)
- [x] Did you make sure to update the documentation with your changes? (if necessary)

## PR review
Anyone in the community is free to review the PR once the tests have passed.

Thank you for contributing to [AutoDL](https://github.com/Auto-DL/Auto-DL). We look forward to your continued support.

### CODE REVIEW

File: scripts/run.sh [hunk 1]
```diff
@@ -129,6 +136,17 @@ terminate(){
 }
 
 setup() {
+
+    echo "************ select the versions of auto-dl ************"
+    echo "  1) Version 1"
+    echo "  2) Version 2"
```
Comment (McTechie) [inline scripts/run.sh:142]:
```suggestion

    echo "************ Select a Version for the Frontend Setup ************"
    echo "  1) Version 1 (React Client)"
    echo "  2) Version 2 (NextJS Client)"
```

File: scripts/run.sh [hunk 2]
```diff
@@ -129,6 +136,17 @@ terminate(){
...
+
+    read n
+    case $n in
+        1) frontend_setup;;
```
Comment (McTechie) [inline scripts/run.sh:146]:
```suggestion
        1) frontend_setup_v1;;
```

File: scripts/run.sh [hunk 3]
```diff
@@ -129,6 +136,17 @@ terminate(){
...
+        2) frontend_setup_v2;;
+        *) echo "invalid option";;
+    esac
     echo -e "${INFO} Setting up..."
     frontend_setup
```
Comment (McTechie) [inline scripts/run.sh:151]:
```suggestion
```

**Review by McTechie — CHANGES_REQUESTED** [review]
Good work! Just a few changes would be required xD

**Comment by ashish-ad** [comment]
Hey @McTechie, I have merged all the changes as u suggested and _I wanted to learn how do you do this, I mean the suggestion of changes but suggesting certain lines to be changed because I too want to learn this as I am too a collaborator maintainer I too want to learn this._
And I think the branch is ready to get merged in the main branch.

**Comment by McTechie** [comment]
> Hey @McTechie, I have merged all the changes as u suggested and _I wanted to learn how do you do this, I mean the suggestion of changes but suggesting certain lines to be changed because I too want to learn this as I am too a collaborator maintainer I too want to learn this._ And I think the branch is ready to get merged in the main branch.

@ashish-ad, great work! The script looks good to me, although there would be another review required prior to merging this PR. 

As for suggesting changes, all you gotta do is select the '+' icon next to the line you want to suggest a change following which you would select the '+-' icon shown on a tab above the textarea. If this sounds too confusing, going through [this guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-proposed-changes-in-a-pull-request#starting-a-review) would be very beneficial. 😄✨

**Comment by ashish-ad** [comment]
Thank you @McTechie.

**Review by RusherRG — APPROVED** [review]

**Comment by ashish-ad** [comment]
Now is this branch ready to merge ?!

**Comment by ashish-ad** [comment]
Hello @RusherRG, Will, this PR will be getting merged, or do I need to do some more changes to it.
please let me know.

**Comment by RusherRG** [comment]
> Hello @RusherRG, Will, this PR will be getting merged, or do I need to do some more changes to it. please let me know.

Hey sorry for the delay. Don't worry we'll merge it, there's no need for any changes.

**Pull request merged**